### ISSUE TITLE
DB keeps going away PDNS 4.0.6

### ISSUE THREAD

**Issue by dac2020** [PowerDNS/pdns#7536]
I searched before posting this and can only find old posts. Since this seems to have been a problem for many years in different variations, I cannot seem to find a resolution so thought I would ask here.

I am running 

pdns 4.0.6-1pdns.el7 , pdns-backend-mysql 4.0.6-1pdns.el7, pdns-recursor 3.7.4-1.el7 on a Centos 7.6 server. 

The server also has the following packages installed. 
mariadb 5.5.60-1.el7_5, mariadb-libs 5.5.60-1.el7_5, perl-DBD-MySQL 4.023-6.el7
php-mysqlnd 5.4.16-46.el7, sqlite 3.7.17-8.el7

For many weeks now, I see something in the logs that I cannot seem to fix, some sort of DB connection error. It seems to reset itself over and over, presumably a process in pdns that checks for a db connection now and then.

Here are some of the errors that show when this happens.
`
Mar 01 09:56:39 ns1.xxx.com systemd[1]: Started PowerDNS Authoritative Server.
Mar 01 09:57:39 ns1.xxx.com pdns_server[23085]: Mar 01 09:57:39 Exiting because communicator thread died with error: GSQLBackend unable to retrieve list of master domains: Could not execute mysql statement: select id,name,m   aster,last_check,notified_serial,type from domains where type='MASTER': Lost connection to MySQL server during query

Another...

Started PowerDNS Authoritative Server.
Backend reported permanent error which prevented lookup 
GSQLBackend lookup query
Could not prepare statement: SELECT 
content,ttl,prio,type,domain_id,disabled,name,auth FROM records WHERE disabled=0 and type=? and name=?
MySQL server has gone away
aborting

Another...

TCP nameserver had error, cycling backend
GSQLBackend lookup query
Could not prepare statement: SELECT 
content,ttl,prio,type,domain_id,disabled,name,auth FROM records WHERE disabled=0 and type=? and name=?
MySQL server has gone away
TCP server is without backend connections, launching
`
The system is dedicated to this purpose only, nothing else running on it.
Does anyone know where I should be looking for leads on fixing this? 

Thanks.

**Comment by pieterlexis** [comment]
Please check the state of the MySQL server process and its logs.

I am closing this issue at it appears a support question and not a bug or issue. If you require further support, please post your issue with all configuration information and logs to the [pdns-users mailinglist](https://mailman.powerdns.com/mailman/listinfo/pdns-users), or join us on [IRC](ircs://irc.oftc.net:6697/#powerdns).

**Comment by dac2020** [comment]
This doesn't seem to be a support issue but something with pdns.
There is nothing wrong with the DB service, at least there is nothing in the logs showing anything unusual. 
I've checked it, it's always up and running. It is pdns that seems to think there is a problem reaching the db.

**Comment by ahupowerdns** [comment]
It actually HAS a problem reaching the db. We don't make up this error. You may want to look into your mysql timeout settings.

**Comment by dac2020** [comment]
Yes, I saw a lot of posts about that so added timeout settings which does not seem to have changed anything. I'll take another look. Maybe there is something I missed.

**Comment by dac2020** [comment]
It makes no sense.
I've added more memory to the server just to see if things would change. Nothing.
I added connect_timeout = 40 and wait_timeout=40 and still no change.

This server was working perfectly until an update a while back. 

Mar 02 09:18:50 ns1.xxx.com pdns_server[5848]: Mar 02 09:18:50 Exiting because communicator thread died with error: GSQLBackend unable to retrieve list of master domains: Could not execute mysql statement: select id,name,master,last_check,notified_serial,type from domains where type='MASTER': Lost connection to MySQL server during query

Those posting here can absolutely confirm that this is not a PDNS issue then? I mean, PDNS installed the mysql server as well during the installation if I recall right. Is it using the /etc/my.cnf or another somewhere else?

**Comment by Habbie** [comment]
> I added connect_timeout = 40 and wait_timeout=40 and still no change.

That won't help - you should try way bigger numbers, on the order of days, if this is a very quiet pdns install.

**Comment by dac2020** [comment]
Correct, it's not high volume server at all. 

I noticed I cannot modify records from the GUI so assume this is because of the DB error.

Error: Failed to call pdnssec utility.
Error: Failed to deactivate DNSSEC keys.

I also ran
pdnsutil check-all-zones

And found two errors. Maybe this is what is causing the problem. I don't know. Still reading countless forum posts and getting nowhere.

'No NS record at zone apex in zone' arpa
